bundles/org.eclipse.jface/src/org/eclipse/jface/resource/URLImageDescriptor.java

```diff
@@ -147,8 +149,10 @@ private static ImageData getImageData(URL url, int fileZoom, int targetZoom) {
 	}
 
 	@SuppressWarnings("restriction")
-	private static ImageData loadImageFromStream(InputStream stream, int fileZoom, int targetZoom) {
-		return NativeImageLoader.load(new ElementAtZoom<>(stream, fileZoom), new ImageLoader(), targetZoom).get(0)
+	private static ImageData loadImageFromStream(InputStream stream, int fileZoom, int targetZoom,
+			Supplier<Point> hintProvider) {
+		return NativeImageLoader
```

Comment:
NativeImageLoader already provides a method for loading images with a target width/height, which works for SVGs, couldnt this be used?

I am not clear why the zoom and the desired size both should be passed down, and then is the expectation that both this values should be always consistent. Wouldnt it make sense to pass the target size for svgs as query params and use zoom based `NativeImageLoader::load `method if the query params are null?